dali/ft/ftslavelib.cpp

```diff
@@ -0,0 +1,133 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jliball.hpp"
+#include "platform.h"
+#include "jlib.hpp"
+#include "jio.hpp"
+#include "jmutex.hpp"
+#include "jfile.hpp"
+#include "jsocket.hpp"
+#include "jdebug.hpp"
+
+#include "fterror.hpp"
+#include "dadfs.hpp"
+#include "rmtspawn.hpp"
+#include "filecopy.hpp"
+#include "fttransform.ipp"
+#include "daftformat.hpp"
+#include "daftcfg.hpp"
+#include "mptag.hpp"
+
+bool processPullCommand(ISocket * masterSocket, MemoryBuffer & msg)
+{
+    srand((int)get_cycles_now());
+    TransferServer server(masterSocket);
+    server.deserializeAction(msg, FTactionpull);
+    return server.pull();
+}
+
+
+bool processPushCommand(ISocket * masterSocket, MemoryBuffer & msg)
+{
+    srand((int)get_cycles_now());
+    TransferServer server(masterSocket);
+    server.deserializeAction(msg, FTactionpush);
+    return server.push();
+}
+
+
+bool processPartitionCommand(ISocket * masterSocket, MemoryBuffer & msg, MemoryBuffer & results)
+{
+    FileFormat srcFormat;
+    FileFormat tgtFormat;
+    unsigned whichInput;
+    RemoteFilename fullPath;
+    offset_t totalSize;
+    offset_t thisOffset;
+    offset_t thisSize;
+    unsigned thisHeaderSize;
+    unsigned numParts;
+    bool compressedInput = false;
+    unsigned compatflags = 0;  
+
+    srcFormat.deserialize(msg);
+    tgtFormat.deserialize(msg);
+    msg.read(whichInput);
+    fullPath.deserialize(msg);
+    msg.read(totalSize);
+    msg.read(thisOffset);
+    msg.read(thisSize);
+    msg.read(thisHeaderSize);
+    msg.read(numParts);
+    if (msg.remaining())
+        msg.read(compressedInput);
+    if (msg.remaining())
+        msg.read(compatflags); // not yet used
+    StringAttr decryptkey;
+    if (msg.remaining())
+        msg.read(decryptkey);
+    if (msg.remaining())
+    {
+        srcFormat.deserializeExtra(msg, 1);
+        tgtFormat.deserializeExtra(msg, 1);
+    }
+
+    StringBuffer text;
+    fullPath.getRemotePath(text);
+    LOG(MCdebugProgress, unknownJob, "Process partition %d(%s)", whichInput, text.str());
+    Owned<IFormatProcessor> processor = createFormatProcessor(srcFormat, tgtFormat, true);
+    Owned<IOutputProcessor> target = createOutputProcessor(tgtFormat);
+    processor->setTarget(target);
+
+    processor->setPartitionRange(totalSize, thisOffset, thisSize, thisHeaderSize, numParts);
+    processor->setSource(whichInput, fullPath, compressedInput, decryptkey);
+    processor->calcPartitions(NULL);
+
+    PartitionPointArray partition;
+    processor->getResults(partition);
+
+    serialize(partition, results);
+    return true;
+}
+
+bool processFtCommand(byte action, ISocket * masterSocket, MemoryBuffer & msg, MemoryBuffer & results)
```

Comment:
Missing `extern FTSLAVELIB_API` - also will need a `#include "ftslavelib.hpp"`

Reply:
we don't normally add the declspec macro to the cpp implementation,
but the cpp would normally include the header along with the declaration of exports with their declspec's.
Adding the hpp has resolved the problem.